src/components/visit/visitInstance.tsx

```diff
@@ -0,0 +1,241 @@
+                  event.preventDefault()
+                }}
+              >
+                <div className='font-bold peer-checked:font-normal'>
+                  <input
+                    placeholder='Date'
+                    value={this.state.dateTime}
+                    onChange={(event) =>
+                      this.setState({ dateTime: event.target.value })
+                    }
+                    className='bg-gray font-normal text-primary'
+                  />
+                  <div>
+                    <input
+                      size={8}
+                      className='bg-gray text-sm font-normal text-primary'
+                      placeholder='Last Name'
+                      value={this.state.lastName}
+                      onChange={(event) =>
+                        this.setState({ lastName: event.target.value })
+                      }
+                    />
+                    <input
+                      size={8}
+                      className='bg-gray text-sm font-normal text-primary'
+                      placeholder='First Name'
+                      value={this.state.firstName}
+                      onChange={(event) =>
+                        this.setState({ firstName: event.target.value })
+                      }
+                    />
+                  </div>
+                </div>
+                <p className='text-sm'>
+                  Pet/Pets:{' '}
+                  <input
+                    size={8}
+                    className='bg-gray text-primary'
+                    placeholder='Pet Name(s)'
+                    value={this.state.petName}
+                    onChange={(event) =>
+                      this.setState({ petName: event.target.value })
+                    }
+                  />
+                </p>
+                <p className='text-sm'>
+                  Duration:{' '}
+                  <input
+                    size={8}
+                    className='bg-gray text-primary'
+                    placeholder='Duration'
+                    value={this.state.duration}
+                    onChange={(event) =>
+                      this.setState({ duration: event.target.value })
+                    }
+                  />
+                </p>
+                <p className='text-sm'>
+                  Walk Metres:{' '}
+                  <input
+                    size={8}
+                    className='bg-gray p-0 text-sm text-primary'
+                    type='number'
+                    step='0.001'
+                    placeholder='Distance'
+                    value={this.state.walkDist.toString()}
+                    onChange={(event) => {
+                      if (!isNaN(parseFloat(event.target.value))) {
+                        this.setState({
+                          walkDist: parseFloat(event.target.value)
+                        })
+                      } else {
+                        this.setState({ walkDist: 0 })
```

Comment:
These functions are a bit complex and might be worth extracting out of your display logic.